server/graphql/schema.js

```diff
@@ -55,46 +54,72 @@ const typeDefs = gql`
 `;
 
 const resolvers = {
+  // query resolvers
   Query: {
+    // fetch a user by ID
     user: async (_, args) => {
       return await User.findById(args.id);
     },
+    // fetch messages for a specific chatroom
     messages: async (_, { chatroomId }) => {
       return await Message.find({ location: chatroomId });
     },
+    // fetch all chatrooms
     chatrooms: async () => {
       return await Chatroom.find({}).populate('tags');
     },
   },
+  // mutation resolvers
   Mutation: {
-    login: async (_, { email, password }, context) => {
+    // user login
+    LOGIN_USER: async (_, { email, password }, context) => {
       const user = await User.findOne({ email });
       if (!user || !await user.isCorrectPassword(password)) {
         throw new Error('Invalid credentials');
       }
       const token = // jwt token here
       return { token, user };
     },
-    addUser: async (_, { username, email, password }, context) => {
+    // add a new user
+    ADD_USER: async (_, { username, email, password }, context) => {
       const newUser = new User({ username, email, password });
       await newUser.save();
       const token = // jwt token here
       return { token, user: newUser };
     },
-    updateUser: async (_, { username, email, password }, context) => {
+    // update an existing user
+    UPDATE_USER: async (_, { username, email, password }, context) => {
       const updatedUser = await User.findOneAndUpdate(
         { username },
         { email, password },
         { new: true }
       );
       return updatedUser;
     },
-    addMessage: async (_, { sender, content, thread, location }, context) => {
+     // add a new message
+    ADD_MESSAGE: async (_, { sender, content, thread, location }, context) => {
       const newMessage = new Message({ sender, content, thread, location });
       await newMessage.save();
       return newMessage;
     },
-    addChatroom: async (_, { title, tagIds, icon }, context) => {
+    // update an existing message
+    UPDATE_MESSAGE: async (_, { messageId, content }, context) => {
+      if (!context.user) {
+        throw new Error('Authentication required');
+      }
+      const message = await Message.findById(messageId);
+      if (!message) {
+        throw new Error('Message not found');
+      }
+      if (message.sender.toString() !== context.user._id.toString()) {
```

Comment:
Interesting addition here: it makes our data validation safer, if marginally less performant (I don't believe we care about a few extra function calls)